tests/e2e/note-images.test.ts

```diff
@@ -0,0 +1,154 @@
+	)
+	await page
+		.locator('[id="note-editor-images\\[1\\]\\.altText"]')
+		.fill(altText2)
+
+	await page.getByRole('button', { name: 'submit' }).click()
+	await expect(page).toHaveURL(new RegExp(`/users/${user.username}/notes/.*`))
+	await expect(page.getByRole('heading', { name: newNote.title })).toBeVisible()
+	await expect(page.getByAltText(altText1)).toBeVisible()
+	await expect(page.getByAltText(altText2)).toBeVisible()
+})
+
+test('Users can edit note image', async ({ page }) => {
+	const user = await loginPage({ page })
+
+	const note = await prisma.note.create({
+		select: { id: true },
+		data: {
+			...createNoteWithImage(),
+			ownerId: user.id,
+		},
+	})
+	await page.goto(`/users/${user.username}/notes/${note.id}`)
+
+	// edit the image
+	await page.getByRole('link', { name: 'Edit', exact: true }).click()
+	const updatedImage = {
+		altText: 'koala coder',
+		url: 'tests/fixtures/images/kody-notes/koala-coder.png',
+	}
+	await page.setInputFiles("input[name='images[0].file']", updatedImage.url)
+	await page
+		.locator('[id="note-editor-images\\[0\\]\\.altText"]')
+		.fill(updatedImage.altText)
+	await page.getByRole('button', { name: 'submit' }).click()
+
+	await expect(page).toHaveURL(new RegExp(`/users/${user.username}/notes/.*`))
+	await expect(page.getByAltText(updatedImage.altText)).toBeVisible()
+})
+
+test('Users can delete note image', async ({ page }) => {
+	const user = await loginPage({ page })
+
+	const note = await prisma.note.create({
+		select: { id: true, title: true },
+		data: {
+			...createNoteWithImage(),
+			ownerId: user.id,
+		},
+	})
+	await page.goto(`/users/${user.username}/notes/${note.id}`)
+
+	await expect(page.getByRole('heading', { name: note.title })).toBeVisible()
+	// find image tags
+	const images = page
+		.getByRole('main')
+		.getByRole('list')
+		.getByRole('listitem')
+		.getByRole('img')
+	const countBefore = await images.count()
+	await page.getByRole('link', { name: 'Edit', exact: true }).click()
+	await page.getByRole('button', { name: 'remove image' }).click()
+	await page.getByRole('button', { name: 'submit' }).click()
+	await expect(page).toHaveURL(new RegExp(`/users/${user.username}/notes/.*`))
+	const countAfter = await images.count()
+	expect(countAfter).toEqual(countBefore - 1)
+})
+
+function createNote() {
+	return {
+		title: faker.lorem.words(3),
+		content: faker.lorem.paragraphs(3),
+	} satisfies Omit<Note, 'id' | 'createdAt' | 'updatedAt' | 'type' | 'ownerId'>
+}
+function createNoteWithImage() {
+	return {
+		...createNote(),
+		images: {
+			create: {
+				altText: 'cute koala',
+				contentType: 'image/png',
+				blob: fs.readFileSync(
+					'tests/fixtures/images/kody-notes/cute-koala.png',
+				),
+			},
+		},
+	} satisfies Omit<
+		Note,
+		'id' | 'createdAt' | 'updatedAt' | 'type' | 'ownerId'
+	> & {
+		images: { create: Pick<NoteImage, 'altText' | 'blob' | 'contentType'> }
+	}
+}
+
+test.afterEach(async () => {
```

Comment:
Is this necessary? We already handle this in the utils module I think. But it's not working so maybe this is necessary? #415 

I'd love to not require this

Reply:
fixed in https://github.com/epicweb-dev/epic-stack/pull/420/commits/f13d773367c1f63ebc3eedc6808d0422277580a0